framework/test/Volo.Abp.Authorization.Tests/Volo/Abp/Authorization/Authorization_Tests.cs

```diff
@@ -52,5 +52,30 @@ public void Should_Permission_Definition_GetGroup()
         {
             _permissionDefinitionManager.GetGroups().Count.ShouldBe(1);
         }
+
+        [Fact]
+        public async Task Should_Not_Allow_To_Call_Method_If_Has_No_Role_ProtectedByRole_Async()
+        {
+            await Assert.ThrowsAsync<AbpAuthorizationException>(async () =>
+            {
+                await _myAuthorizedService1.ProtectedByRole().ConfigureAwait(false);
+            }).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task Should_Allow_To_Call_Method_If_Has_No_Role_ProtectedByRole_Async()
+        {
+            int result = await _myAuthorizedService1.ProtectedByRole().ConfigureAwait(false);
+            result.ShouldBe(42);
+        }
```

Comment:
```
await _myAuthorizedService1.ProtectedByRole().ConfigureAwait(false);
```
I guess one of the unit test methods will fail.

Reply:
Yes, it is fail. I am still developing for that. Had very little.

Reply:
Hi @maliming ,
I finished finally. You can check.